build_torcx_store

```diff
@@ -151,10 +167,52 @@ function torcx_package() {
                 :  # Set $? to 0 or the pipeline fails and -e quits.
         done
 
-        # Package the installed files.
-        file="${BUILD_DIR}/${name}:${version}.torcx.tgz"
-        tar --force-local -C "${tmproot}" -czf "${file}" .torcx bin lib
-        ln -fns "${file##*/}" "${BUILD_DIR}/${name}:com.coreos.cl.torcx.tgz"
+        # Move anything we plan to package to its root
+        sudo mv "${tmproot}"/{.torcx,bin,lib} "${tmppkgroot}"
+
+        # Create a reproducible digest by which this package will be uploaded
+        # and referred
+        digest=$(casync \
+                        --digest=sha512-256 \
+                        --without=selinux \
+                        --with=flag-noatime \
+                        --what=directory \
+                        digest "${tmppkgroot}")
+
+        pkg_cas_root="${TORCX_CAS_ROOT}/${name}/${digest}"
+        pkg_cas_file="${pkg_cas_root}/${name}:${version}.torcx.tgz"
+
+        # Create the cas store if it doesn't exist
+        if [[ ! -d "${pkg_cas_root}" ]]; then
+                mkdir -p "${TORCX_CAS_ROOT}/${name}"
+                # Package the installed files.
+                tmpcas="${BUILD_DIR}/${digest}"
+                mkdir -p "${tmpcas}"
+                tmpfile="${tmpcas}/${name}:${version}.torcx.tgz"
+                tar --force-local -C "${tmppkgroot}" -czf "${tmpfile}" .
+                # atomically move the whole directory in place from a tmp locatoin
+                mv "${tmpcas}" "${pkg_cas_root}"
+        fi
+        [[ -f "${pkg_cas_file}" ]] || die "${pkg_cas_file} should exist but didn't"
+        sha512sum=$(sha512sum "${pkg_cas_file}" | awk '{print $1}')
+
+        update_default=false
+        if [[ "${type}" == "default" ]]; then
+                update_default=true
+                pkg_locations+=("/usr/share/torcx/store/${name}:${version}.torcx.tgz")
+        fi
+	if [[ "${FLAGS_upload}" -eq ${FLAGS_TRUE} ]]; then
+		pkg_locations+=("$(download_tectonic_torcx_url "pkgs/${BOARD}/${name}/${digest}/${name}:${version}.torcx.tgz")")
+	fi
+        torcx_manifest::add_pkg "${manifest_path}" \
+            "${name}" \
+            "${version}" \
+            "sha512-${sha512sum}" \
```

Comment:
Could the separator be a ':' ? That matches the opencontainers digest format.

Reply:
I think the answer is no, as this matches ignition format instead.

Reply:
Yeah, I'd prefer the arbitrary format ignition chose over the arbitrary format tarsum chose.